lib/std/fmt.zig

```diff
@@ -599,6 +600,16 @@ pub fn formatText(
             try formatInt(c, 16, fmt[0] == 'X', FormatOptions{ .width = 2, .fill = '0' }, writer);
         }
         return;
+    } else if (comptime (std.mem.eql(u8, fmt, "e") or std.mem.eql(u8, fmt, "E"))) {
+        for (bytes) |c| {
+            if (std.ascii.isPrint(c)) {
```

Comment:
Do we consider `\t`, `\n` and `\r` printable? (#4014)

Reply:
For purposes of a debug print: I don't think they should be considered printable:
  - its hard to visually tell the difference between a tab that is one space wide an a normal space character.
  - I imagine this would often be used for line based output, where escaping newlines is desired.
  - `\r` *is* unprintable